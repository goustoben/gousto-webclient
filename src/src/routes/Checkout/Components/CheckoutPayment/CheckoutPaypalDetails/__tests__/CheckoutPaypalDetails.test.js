import React from 'react'
import { shallow } from 'enzyme'

import { CheckoutPaypalDetails } from '../CheckoutPaypalDetails'

describe('CheckoutPaypalDetails', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallow(<CheckoutPaypalDetails hide={false} />)
  })

  test('should render correctly', () => {
    expect(wrapper.hasClass('hide')).toBeFalsy()
  })

  describe('when hide is true', () => {
    beforeEach(() => {
      wrapper.setProps({ hide: true })
    })

    test('then it should be hidden', () => {
      expect(wrapper.hasClass('hide')).toBeTruthy()
    })
  })
})
