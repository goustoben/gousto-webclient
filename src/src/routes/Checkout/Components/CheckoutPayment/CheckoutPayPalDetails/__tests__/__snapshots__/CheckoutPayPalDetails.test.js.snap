// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckoutPayPalDetails should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CheckoutPayPalDetails
    clearPayPalErrors={[Function]}
    firePayPalError={[Function]}
    hasErrors={false}
    hide={false}
    isCheckoutOverhaulEnabled={false}
    isPayPalSetupDone={false}
    isPaymentMethodVariationEnabled={false}
    paypalScriptsReady={false}
    setCurrentPaymentMethod={[Function]}
    setPayPalDeviceData={[Function]}
    setPayPalNonce={[Function]}
    token={null}
    trackEvent={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className=""
        >
          <p
            className="text"
          >
            <span
              className="padlockIcon"
            />
            You will be prompted by PayPal for payment details to securely setup your subscription.
          </p>
          <p
            className="text"
          >
            PayPal is loading...
          </p>
          <div
            className="paypalContainer transparent"
            id="paypal-container"
          />
        </div>,
        false,
      ],
      "className": "",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p
              className="text"
            >
              <span
                className="padlockIcon"
              />
              You will be prompted by PayPal for payment details to securely setup your subscription.
            </p>,
            <p
              className="text"
            >
              PayPal is loading...
            </p>,
            <div
              className="paypalContainer transparent"
              id="paypal-container"
            />,
          ],
          "className": "",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span
                  className="padlockIcon"
                />,
                "You will be prompted by PayPal for payment details to securely setup your subscription.",
              ],
              "className": "text",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "padlockIcon",
                },
                "ref": null,
                "rendered": null,
                "type": "span",
              },
              "You will be prompted by PayPal for payment details to securely setup your subscription.",
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "PayPal is loading...",
              "className": "text",
            },
            "ref": null,
            "rendered": "PayPal is loading...",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "paypalContainer transparent",
              "id": "paypal-container",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
        ],
        "type": "div",
      },
      false,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className=""
          >
            <p
              className="text"
            >
              <span
                className="padlockIcon"
              />
              You will be prompted by PayPal for payment details to securely setup your subscription.
            </p>
            <p
              className="text"
            >
              PayPal is loading...
            </p>
            <div
              className="paypalContainer transparent"
              id="paypal-container"
            />
          </div>,
          false,
        ],
        "className": "",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p
                className="text"
              >
                <span
                  className="padlockIcon"
                />
                You will be prompted by PayPal for payment details to securely setup your subscription.
              </p>,
              <p
                className="text"
              >
                PayPal is loading...
              </p>,
              <div
                className="paypalContainer transparent"
                id="paypal-container"
              />,
            ],
            "className": "",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <span
                    className="padlockIcon"
                  />,
                  "You will be prompted by PayPal for payment details to securely setup your subscription.",
                ],
                "className": "text",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "padlockIcon",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "span",
                },
                "You will be prompted by PayPal for payment details to securely setup your subscription.",
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "PayPal is loading...",
                "className": "text",
              },
              "ref": null,
              "rendered": "PayPal is loading...",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "paypalContainer transparent",
                "id": "paypal-container",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
          ],
          "type": "div",
        },
        false,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;

exports[`CheckoutPayPalDetails when PayPal initialized should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CheckoutPayPalDetails
    clearPayPalErrors={[Function]}
    firePayPalError={[Function]}
    hasErrors={false}
    hide={false}
    isCheckoutOverhaulEnabled={false}
    isPayPalSetupDone={false}
    isPaymentMethodVariationEnabled={false}
    paypalScriptsReady={false}
    setCurrentPaymentMethod={[Function]}
    setPayPalDeviceData={[Function]}
    setPayPalNonce={[Function]}
    token={null}
    trackEvent={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className=""
        >
          <p
            className="text"
          >
            <span
              className="padlockIcon"
            />
            You will be prompted by PayPal for payment details to securely setup your subscription.
          </p>
          <div
            className="paypalContainer"
            id="paypal-container"
          />
        </div>,
        false,
      ],
      "className": "",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p
              className="text"
            >
              <span
                className="padlockIcon"
              />
              You will be prompted by PayPal for payment details to securely setup your subscription.
            </p>,
            false,
            <div
              className="paypalContainer"
              id="paypal-container"
            />,
          ],
          "className": "",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span
                  className="padlockIcon"
                />,
                "You will be prompted by PayPal for payment details to securely setup your subscription.",
              ],
              "className": "text",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "padlockIcon",
                },
                "ref": null,
                "rendered": null,
                "type": "span",
              },
              "You will be prompted by PayPal for payment details to securely setup your subscription.",
            ],
            "type": "p",
          },
          false,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "paypalContainer",
              "id": "paypal-container",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
        ],
        "type": "div",
      },
      false,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className=""
          >
            <p
              className="text"
            >
              <span
                className="padlockIcon"
              />
              You will be prompted by PayPal for payment details to securely setup your subscription.
            </p>
            <div
              className="paypalContainer"
              id="paypal-container"
            />
          </div>,
          false,
        ],
        "className": "",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p
                className="text"
              >
                <span
                  className="padlockIcon"
                />
                You will be prompted by PayPal for payment details to securely setup your subscription.
              </p>,
              false,
              <div
                className="paypalContainer"
                id="paypal-container"
              />,
            ],
            "className": "",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <span
                    className="padlockIcon"
                  />,
                  "You will be prompted by PayPal for payment details to securely setup your subscription.",
                ],
                "className": "text",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "padlockIcon",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "span",
                },
                "You will be prompted by PayPal for payment details to securely setup your subscription.",
              ],
              "type": "p",
            },
            false,
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "paypalContainer",
                "id": "paypal-container",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
          ],
          "type": "div",
        },
        false,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;

exports[`CheckoutPayPalDetails when PayPal setup is done should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CheckoutPayPalDetails
    clearPayPalErrors={[Function]}
    firePayPalError={[Function]}
    hasErrors={false}
    hide={false}
    isCheckoutOverhaulEnabled={false}
    isPayPalSetupDone={true}
    isPaymentMethodVariationEnabled={false}
    paypalScriptsReady={false}
    resetPaymentMethod={[MockFunction]}
    setCurrentPaymentMethod={[Function]}
    setPayPalDeviceData={[Function]}
    setPayPalNonce={[Function]}
    token={null}
    trackEvent={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="hide"
        >
          <p
            className="text"
          >
            <span
              className="padlockIcon"
            />
            You will be prompted by PayPal for payment details to securely setup your subscription.
          </p>
          <p
            className="text"
          >
            PayPal is loading...
          </p>
          <div
            className="paypalContainer transparent"
            id="paypal-container"
          />
        </div>,
        <Connect(PayPalConfirmation) />,
      ],
      "className": "",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p
              className="text"
            >
              <span
                className="padlockIcon"
              />
              You will be prompted by PayPal for payment details to securely setup your subscription.
            </p>,
            <p
              className="text"
            >
              PayPal is loading...
            </p>,
            <div
              className="paypalContainer transparent"
              id="paypal-container"
            />,
          ],
          "className": "hide",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span
                  className="padlockIcon"
                />,
                "You will be prompted by PayPal for payment details to securely setup your subscription.",
              ],
              "className": "text",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "padlockIcon",
                },
                "ref": null,
                "rendered": null,
                "type": "span",
              },
              "You will be prompted by PayPal for payment details to securely setup your subscription.",
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "PayPal is loading...",
              "className": "text",
            },
            "ref": null,
            "rendered": "PayPal is loading...",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "paypalContainer transparent",
              "id": "paypal-container",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="hide"
          >
            <p
              className="text"
            >
              <span
                className="padlockIcon"
              />
              You will be prompted by PayPal for payment details to securely setup your subscription.
            </p>
            <p
              className="text"
            >
              PayPal is loading...
            </p>
            <div
              className="paypalContainer transparent"
              id="paypal-container"
            />
          </div>,
          <Connect(PayPalConfirmation) />,
        ],
        "className": "",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p
                className="text"
              >
                <span
                  className="padlockIcon"
                />
                You will be prompted by PayPal for payment details to securely setup your subscription.
              </p>,
              <p
                className="text"
              >
                PayPal is loading...
              </p>,
              <div
                className="paypalContainer transparent"
                id="paypal-container"
              />,
            ],
            "className": "hide",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <span
                    className="padlockIcon"
                  />,
                  "You will be prompted by PayPal for payment details to securely setup your subscription.",
                ],
                "className": "text",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "padlockIcon",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "span",
                },
                "You will be prompted by PayPal for payment details to securely setup your subscription.",
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "PayPal is loading...",
                "className": "text",
              },
              "ref": null,
              "rendered": "PayPal is loading...",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "paypalContainer transparent",
                "id": "paypal-container",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;

exports[`CheckoutPayPalDetails when hide is true should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CheckoutPayPalDetails
    clearPayPalErrors={[Function]}
    firePayPalError={[Function]}
    hasErrors={false}
    hide={true}
    isCheckoutOverhaulEnabled={false}
    isPayPalSetupDone={false}
    isPaymentMethodVariationEnabled={false}
    paypalScriptsReady={false}
    setCurrentPaymentMethod={[Function]}
    setPayPalDeviceData={[Function]}
    setPayPalNonce={[Function]}
    token={null}
    trackEvent={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className=""
        >
          <p
            className="text"
          >
            <span
              className="padlockIcon"
            />
            You will be prompted by PayPal for payment details to securely setup your subscription.
          </p>
          <p
            className="text"
          >
            PayPal is loading...
          </p>
          <div
            className="paypalContainer transparent"
            id="paypal-container"
          />
        </div>,
        false,
      ],
      "className": "hide",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p
              className="text"
            >
              <span
                className="padlockIcon"
              />
              You will be prompted by PayPal for payment details to securely setup your subscription.
            </p>,
            <p
              className="text"
            >
              PayPal is loading...
            </p>,
            <div
              className="paypalContainer transparent"
              id="paypal-container"
            />,
          ],
          "className": "",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span
                  className="padlockIcon"
                />,
                "You will be prompted by PayPal for payment details to securely setup your subscription.",
              ],
              "className": "text",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "padlockIcon",
                },
                "ref": null,
                "rendered": null,
                "type": "span",
              },
              "You will be prompted by PayPal for payment details to securely setup your subscription.",
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "PayPal is loading...",
              "className": "text",
            },
            "ref": null,
            "rendered": "PayPal is loading...",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "paypalContainer transparent",
              "id": "paypal-container",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
        ],
        "type": "div",
      },
      false,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className=""
          >
            <p
              className="text"
            >
              <span
                className="padlockIcon"
              />
              You will be prompted by PayPal for payment details to securely setup your subscription.
            </p>
            <p
              className="text"
            >
              PayPal is loading...
            </p>
            <div
              className="paypalContainer transparent"
              id="paypal-container"
            />
          </div>,
          false,
        ],
        "className": "hide",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p
                className="text"
              >
                <span
                  className="padlockIcon"
                />
                You will be prompted by PayPal for payment details to securely setup your subscription.
              </p>,
              <p
                className="text"
              >
                PayPal is loading...
              </p>,
              <div
                className="paypalContainer transparent"
                id="paypal-container"
              />,
            ],
            "className": "",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <span
                    className="padlockIcon"
                  />,
                  "You will be prompted by PayPal for payment details to securely setup your subscription.",
                ],
                "className": "text",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "padlockIcon",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "span",
                },
                "You will be prompted by PayPal for payment details to securely setup your subscription.",
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "PayPal is loading...",
                "className": "text",
              },
              "ref": null,
              "rendered": "PayPal is loading...",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "paypalContainer transparent",
                "id": "paypal-container",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
          ],
          "type": "div",
        },
        false,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
